nmattia/niv
# New update mechanism

Fixes #36 .

Got a few things to clean up, but this will greatly facilitate the move to multiple fetchers, and fixes a few bugs.

* Introduces the `Update`, which takes care of lazy loading the IO actions (not run if never read or if value will eventually be discarded)
* Adds a new (unit) test infrastructure
* Moves from `String` to `Text`
* A bit of renaming

Few things left to iron out:
* Re-gen the silly svg
* Check the error handling
* Remove all occurrences of `foo`, `bar`, `baz`, etc
* Clarify the separation between the "CLI" spec and the existing spec (i.e. clarify by changing the variable names)
* Document the hell out of everything

CC @nlewo @michaelpj
